rdrf/rdrf/models/definition/models.py

```diff
@@ -1647,10 +1647,48 @@ def clean(self):
         if num_defaults == 0 and not self.is_default:
             raise ValidationError("One Context Form Group must be chosen as the default")
 
-        if self.naming_scheme == "C" and self.naming_cde_to_use is None:
-            raise ValidationError("Naming cde to use cannot be empty")
+        if self.naming_scheme == "C" and self._valid_naming_cde_to_use(self.naming_cde_to_use) is None:
+            raise ValidationError("Invalid naming cde: Should be form name/section code/cde code where all codes must exist")
 
 
+    def _valid_naming_cde_to_use(self, naming_cde_to_use):
+        validation_message = "Invalid naming cde: Should be form name/section code/cde code where all codes must exist"
+        if naming_cde_to_use:
+            try:
+                naming_cde_expression = naming_cde_to_use.split("/")
+                if len(naming_cde_expression) != 3:
+                    raise ValidationError(validation_message)
+                form_name, section_code, cde_code = naming_cde_expression
+            except :
```

Comment:
too broad exception clause - just trap relevant error - probably a ValueError ( check in python console by attempting to destructure a 3 element tuple from None, say

Reply:
Now capturing ValueError while destructuring a tuple

Reply:
Looks good now